bot/admin/server/src/main/kotlin/service/ObservabilityService.kt

```diff
@@ -62,8 +64,15 @@ object ObservabilityService {
     fun deleteConfig(namespace: String, botId: String) {
         val observabilityConfig = observabilityConfigurationDAO.findByNamespaceAndBotId(namespace, botId)
             ?: WebVerticle.badRequest("No Observability configuration is defined yet [namespace: $namespace, botId: $botId]")
+
         logger.info { "Deleting the Observability Configuration [namespace: $namespace, botId: $botId]" }
-        return observabilityConfigurationDAO.delete(observabilityConfig._id)
+        observabilityConfigurationDAO.delete(observabilityConfig._id)
+
+        val setting = observabilityConfig.setting
+        setting.takeIf { it is LangfuseObservabilitySetting }?.let {
+            logger.info { "Deleting the Observability secret ..." }
+            SecurityUtils.deleteSecret((setting as LangfuseObservabilitySetting).secretKey)
+        } ?: logger.info { "No secret to delete for the current setting." }
```

Comment:
To have a more generic approach using kotlin's reflection APIs, something like this : 

```kotlin
import kotlin.reflect.full.memberProperties
import ai.tock.shared.security.key.SecretKey

fun getSecretKeyAttributes(data: Any): List<SecretKey> {
    return data::class.memberProperties
        .filter { it.returnType.classifier == SecretKey::class } // Filter properties of type SecretKey
        .mapNotNull { it.getter.call(data) as? SecretKey } // Safely get the value as SecretKey
}
```

And apply the deletion on it ... it could be some kind of helper used in all setting ?

Reply:
Ok for this implementation, can you make this change ?